src/notes/FluxNotesEditor.jsx

```diff
@@ -67,6 +68,31 @@ const initialEditorState = {
 };
 
 class FluxNotesEditor extends React.Component {
+    openCompletionPortal = (completionComponentShortcut) => {
+        // Always make sure we use an array here; doesn't always return an array from getValueSelectionOptions
+        const portalOptions = !Lang.isEmpty(completionComponentShortcut.getValueSelectionOptions()) ? completionComponentShortcut.getValueSelectionOptions() : [];
+        this.setState({
+            completionComponentShortcut: completionComponentShortcut,
+            portalOptions: portalOptions,
+            completionComponent: completionComponentShortcut.completionComponent,
+            openedPortal: "CompletionPortal",
+        });
+    }
+
+    closeCompletionPortal = () => {
+        // Clean up some variables stored at the Editor level
+        this.setState({
+            completionComponentShortcut: null,
+            portalOptions: [],
+            completionComponent: null,
+            openedPortal: null,
+        });
+    }
+
+    getCompletionComponent = () => {
```

Comment:
I'd like to see this method go away

Reply:
So could we talk about renaming the function? We need a way to get a direct reference to the component so we can access the onKeyDown function, so we can't get read of this component per se.

Reply:
I'm looking at it more now. My comment was not about the name; it was more about that I'd like to see us not need that function.... maybe that's wishful thinking.

Reply:
Can we add a render method to the CompletionPortalPlugin that renders and can provide a reference to the completion component?

Reply:
Originally the short answer seemed to be no - the primary reason that I previously thought the component needed to be rendered at the FNE level was because FNE needs to give CompletionPortalPlugin some way to reference the `ref` for the actual completionComponent. I now think that this can be done using some slightly more advanced react code, so I will spend some time tomorrow morning looking into this.

Reply:
Let's just leave it as is Dylan. What you have works and is not bad. Thanks for entertaining other options. Maybe React 16 can make this more elegant...